development/Dockerfile

```diff
@@ -1,44 +1,71 @@
+
+# Accepts a desired Nautobot version as build argument, default to 1.2.0
 ARG NAUTOBOT_VER="1.4.0"
-ARG PYTHON_VER=3.7
 
-FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
+# Accepts a desired Python version as build argument, default to 3.7
+ARG PYTHON_VER="3.7"
 
-# Make the value available after the FROM directive
-ARG NAUTOBOT_VER
-ENV prometheus_multiproc_dir=/prom_cache
+# Retreive published development image of Nautobot base which should include most CI dependencies
+FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
 
+# Runtime argument and environment setup
 ARG NAUTOBOT_ROOT=/opt/nautobot
 
+ENV prometheus_multiproc_dir=/prom_cache
 ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
 
-WORKDIR $NAUTOBOT_ROOT
-
-# Configure poetry
+# Don't need virtual environments in container, parallelized installer has inconsistent behavior
 RUN poetry config virtualenvs.create false \
     && poetry config installer.parallel false
 
+# !!! CAUTION MODIFYING LINES ABOVE
 # -------------------------------------------------------------------------------------
-# Install Nautobot Plugin
+# Plugin-specifc system build/test dependencies.
+# 
+# Example: LDAP requires `libldap2-dev` to be apt-installed before the Python package.
 # -------------------------------------------------------------------------------------
-WORKDIR /tmp/install
+# --> Start safe to modify section
 
-# Copy in only pyproject.toml/poetry.lock to help with caching this layer if no updates to dependencies
-COPY poetry.lock pyproject.toml /tmp/install/
+# Uncomment the line below if you are apt-installing any package.
+# RUN apt update
+# RUN apt install libldap2-dev
 
-# Add the requested Nautobot version to pyproject
-# to install the correct version based on the NAUTOBOT_VER argument
-# Otherwise Poetry will override the version in this container
-# with the one in the poetry.lock
-RUN poetry add nautobot=${NAUTOBOT_VER}
-
-# --no-root declares not to install the project package since we're wanting to take advantage of caching dependency installation
-# and the project is copied in and installed after this step
-RUN poetry install --no-interaction --no-ansi --no-root
+# --> Stop safe to modify section
+# -------------------------------------------------------------------------------------
+# Install Nautobot Plugin
+# -------------------------------------------------------------------------------------
+# !!! CAUTION MODIFYING LINES BELOW
 
-# Copy in the rest of the source code and install local Nautobot plugin
+# Copy in the source code
 WORKDIR /source
 COPY . /source
-RUN cp /tmp/install/* /source/
-RUN poetry install --no-interaction --no-ansi
+
+# Get container's installed Nautobot version as a forced constraint
+RUN pip show nautobot | grep "Version: " | sed -e 's/Version: /nautobot==/' > constraints.txt
+
+# Use Poetry to grab dev dependencies from the lock file
+# Can be improved in Poetry 1.2 which allows `poetry install --only dev`
```

Comment:
Poetry 1.2 should be our standard at this point. Can we go ahead and implement this improvement?

Reply:
Ah. I hadn't realized we're using the version of Poetry that's packaged into the Docker image. That's not ideal certainly.

Reply:
For what it's worth I did test `poetry install --only dev` and it does seem to honor the installed version of Nautobot even though not explicitly told to do so:

I updated my lockfile to target 1.5.0 and after running the dev dependency install it kept the version as the one provided by the container

```
nautobot-plugin-golden-config on  bsc-2023-upstream-testing [!]
➜ cat poetry.lock | grep "name = \"nautobot\"" -C3
typing-extensions = ">=4.3.0"

[[package]]
name = "nautobot"
version = "1.5.0"
description = "Source of truth and network automation platform."
category = "main"

nautobot-plugin-golden-config on  bsc-2023-upstream-testing [!]
➜ cat pyproject.toml | grep "nautobot ="
nautobot = ">=1.4.0"

nautobot-plugin-golden-config on  bsc-2023-upstream-testing [!]
➜ invoke build
Building Nautobot with Python 3.7...
Running docker-compose command "build"
#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 3.48kB done
#2 DONE 0.0s

#3 [internal] load metadata for ghcr.io/nautobot/nautobot-dev:1.4.2-py3.7  # <======= Requested container version ==========================================================
#3 DONE 0.4s

#4 [ 1/11] FROM ghcr.io/nautobot/nautobot-dev:1.4.2-py3.7@sha256:688a5c8d65ea5be4a98bbe7c5f18ec17af325b31bd38ea4446c7dd81a0049ff3
#4 resolve ghcr.io/nautobot/nautobot-dev:1.4.2-py3.7@sha256:688a5c8d65ea5be4a98bbe7c5f18ec17af325b31bd38ea4446c7dd81a0049ff3 0.0s done
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 46.95kB 0.0s done
#5 DONE 0.0s

#6 [ 2/11] RUN poetry --version
#6 CACHED

#7 [ 3/11] RUN poetry config virtualenvs.create false     && poetry config installer.parallel false
#7 CACHED

#8 [ 4/11] WORKDIR /source
#8 CACHED

#9 [ 5/11] COPY . /source
#9 DONE 0.2s

#10 [ 6/11] RUN pip show nautobot | grep "^Version: " | sed -e 's/Version: /nautobot==/' > constraints.txt
#10 DONE 1.0s

#11 [ 7/11] RUN cat constraints.txt
#11 0.220 nautobot==1.4.2  # <======= Contents of constraint file ==========================================================
#11 DONE 0.2s

#12 [ 8/11] RUN pip install -c constraints.txt -e .
#12 0.506 Obtaining file:///source
#12 0.508   Installing build dependencies: started
#12 3.409   Installing build dependencies: finished with status 'done'
#12 3.410   Checking if build backend supports build_editable: started
#12 3.456   Checking if build backend supports build_editable: finished with status 'done'
#12 3.457   Getting requirements to build editable: started
#12 3.503   Getting requirements to build editable: finished with status 'done'
#12 3.504   Preparing editable metadata (pyproject.toml): started
#12 3.651   Preparing editable metadata (pyproject.toml): finished with status 'done'
#12 3.669 Requirement already satisfied: nautobot>=1.4.0 in /usr/local/lib/python3.7/site-packages (from nautobot-golden-config==1.2.2) (1.4.2)
#12 3.763 Collecting deepdiff<6.0.0,>=5.5.0
#12 3.853   Downloading deepdiff-5.8.1-py3-none-any.whl (69 kB)
#12 3.872      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 69.5/69.5 kB 3.8 MB/s eta 0:00:00
#12 3.894 Collecting django-pivot<2.0.0,>=1.8.1
#12 3.911   Downloading django_pivot-1.9.0-py3-none-any.whl (14 kB)
#12 4.035 Collecting matplotlib<4.0.0,>=3.3.2
#12 4.057   Downloading matplotlib-3.5.3-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (11.6 MB)
#12 4.722      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.6/11.6 MB 19.2 MB/s eta 0:00:00
#12 4.760 Collecting nautobot-plugin-nornir>=1.0.0
#12 4.776   Downloading nautobot_plugin_nornir-1.0.0-py3-none-any.whl (21 kB)
#12 4.804 Collecting ordered-set<4.2.0,>=4.1.0
#12 4.819   Downloading ordered_set-4.1.0-py3-none-any.whl (7.6 kB)
#12 4.823 Requirement already satisfied: django>=2.2.0 in /usr/local/lib/python3.7/site-packages (from django-pivot<2.0.0,>=1.8.1->nautobot-golden-config==1.2.2) (3.2.15)
#12 4.872 Collecting kiwisolver>=1.0.1
#12 4.891   Downloading kiwisolver-1.4.4-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (1.4 MB)
#12 4.946      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.4/1.4 MB 26.3 MB/s eta 0:00:00
#12 4.948 Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.7/site-packages (from matplotlib<4.0.0,>=3.3.2->nautobot-golden-config==1.2.2) (21.3)
#12 4.949 Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.7/site-packages (from matplotlib<4.0.0,>=3.3.2->nautobot-golden-config==1.2.2) (9.2.0)
#12 5.112 Collecting numpy>=1.17
#12 5.140   Downloading numpy-1.21.6-cp37-cp37m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (13.0 MB)
#12 5.943      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.0/13.0 MB 16.1 MB/s eta 0:00:00
#12 5.986 Collecting cycler>=0.10
#12 6.000   Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)
#12 6.004 Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.7/site-packages (from matplotlib<4.0.0,>=3.3.2->nautobot-golden-config==1.2.2) (3.0.9)
#12 6.053 Collecting fonttools>=4.22.0
#12 6.081   Downloading fonttools-4.38.0-py3-none-any.whl (965 kB)
#12 6.136      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 965.4/965.4 kB 18.2 MB/s eta 0:00:00
#12 6.140 Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.7/site-packages (from matplotlib<4.0.0,>=3.3.2->nautobot-golden-config==1.2.2) (2.8.2)
#12 6.163 Requirement already satisfied: graphene-django-optimizer<0.9.0,>=0.8.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.8.0)
#12 6.164 Requirement already satisfied: drf-spectacular[sidecar]<0.23.0,>=0.22.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.22.1)
#12 6.164 Requirement already satisfied: graphene-django<2.16.0,>=2.15.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.15.0)
#12 6.165 Requirement already satisfied: importlib-metadata<5.0,>=4.4 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.12.0)
#12 6.166 Requirement already satisfied: django-ajax-tables<1.2.0,>=1.1.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.1.1)
#12 6.167 Requirement already satisfied: djangorestframework<3.14.0,>=3.13.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.13.1)
#12 6.167 Requirement already satisfied: django-cacheops<6.1,>=6.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (6.0)
#12 6.168 Requirement already satisfied: django-jinja<2.11.0,>=2.10.2 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.10.2)
#12 6.168 Requirement already satisfied: celery<5.3.0,>=5.2.7 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.2.7)
#12 6.169 Requirement already satisfied: Markdown<3.4.0,>=3.3.7 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.3.7)
#12 6.169 Requirement already satisfied: django-extensions<3.3.0,>=3.2.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.2.0)
#12 6.170 Requirement already satisfied: PyYAML<6.1,>=6.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (6.0)
#12 6.170 Requirement already satisfied: GitPython<3.2.0,>=3.1.27 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.1.27)
#12 6.171 Requirement already satisfied: django-filter<21.2,>=21.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (21.1)
#12 6.171 Requirement already satisfied: jsonschema<4.8.0,>=4.7.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.7.2)
#12 6.172 Requirement already satisfied: django-mptt<0.14.0,>=0.13.4 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.13.4)
#12 6.172 Requirement already satisfied: pycryptodome<3.14.0,>=3.13.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.13.0)
#12 6.173 Requirement already satisfied: django-taggit<1.6.0,>=1.5.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.5.1)
#12 6.174 Requirement already satisfied: netutils<1.3.0,>=1.2.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.2.0)
#12 6.174 Requirement already satisfied: pyuwsgi<2.1.0,>=2.0.20 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.0.20)
#12 6.175 Requirement already satisfied: django-db-file-storage<0.6.0,>=0.5.5 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.5.5)
#12 6.176 Requirement already satisfied: svgwrite<1.5.0,>=1.4.2 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.4.3)
#12 6.176 Requirement already satisfied: django-timezone-field<4.2.0,>=4.1.2 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.1.2)
#12 6.177 Requirement already satisfied: django-cors-headers<3.14.0,>=3.13.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.13.0)
#12 6.177 Requirement already satisfied: Jinja2<3.1.0,>=3.0.3 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.0.3)
#12 6.178 Requirement already satisfied: social-auth-app-django<5.1.0,>=5.0.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.0.0)
#12 6.178 Requirement already satisfied: django-cryptography<1.1,>=1.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.0)
#12 6.179 Requirement already satisfied: django-webserver<1.3.0,>=1.2.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.2.0)
#12 6.179 Requirement already satisfied: MarkupSafe<2.2.0,>=2.1.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.1.1)
#12 6.179 Requirement already satisfied: django-rq<2.6.0,>=2.5.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.5.1)
#12 6.180 Requirement already satisfied: django-celery-beat<2.3.0,>=2.2.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.2.1)
#12 6.180 Requirement already satisfied: django-health-check<3.17.0,>=3.16.5 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.16.5)
#12 6.181 Requirement already satisfied: django-tree-queries<0.12,>=0.11 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.11.0)
#12 6.181 Requirement already satisfied: django-constance[database]<2.9.0,>=2.8.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.8.0)
#12 6.182 Requirement already satisfied: drf-yasg[validation]<2.0.0,>=1.20.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.21.3)
#12 6.182 Requirement already satisfied: django-prometheus<2.3.0,>=2.2.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.2.0)
#12 6.182 Requirement already satisfied: psycopg2-binary<2.10.0,>=2.9.3 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.9.3)
#12 6.183 Requirement already satisfied: django-redis<5.3.0,>=5.2.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.2.0)
#12 6.183 Requirement already satisfied: netaddr<0.9.0,>=0.8.0 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.8.0)
#12 6.183 Requirement already satisfied: django-tables2<2.5.0,>=2.4.1 in /usr/local/lib/python3.7/site-packages (from nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.4.1)
#12 6.211 Collecting nornir-nautobot<3.0.0,>=2.2.0
#12 6.226   Downloading nornir_nautobot-2.3.0-py3-none-any.whl (15 kB)
#12 6.277 Requirement already satisfied: kombu<6.0,>=5.2.3 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.2.4)
#12 6.277 Requirement already satisfied: click-repl>=0.2.0 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.2.0)
#12 6.278 Requirement already satisfied: billiard<4.0,>=3.6.4.0 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.6.4.0)
#12 6.278 Requirement already satisfied: click-plugins>=1.1.1 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.1.1)
#12 6.279 Requirement already satisfied: click<9.0,>=8.0.3 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (8.1.3)
#12 6.279 Requirement already satisfied: vine<6.0,>=5.0.0 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.0.0)
#12 6.279 Requirement already satisfied: pytz>=2021.3 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2022.1)
#12 6.280 Requirement already satisfied: click-didyoumean>=0.0.3 in /usr/local/lib/python3.7/site-packages (from celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.3.0)
#12 6.285 Requirement already satisfied: sqlparse>=0.2.2 in /usr/local/lib/python3.7/site-packages (from django>=2.2.0->django-pivot<2.0.0,>=1.8.1->nautobot-golden-config==1.2.2) (0.4.2)
#12 6.285 Requirement already satisfied: asgiref<4,>=3.3.2 in /usr/local/lib/python3.7/site-packages (from django>=2.2.0->django-pivot<2.0.0,>=1.8.1->nautobot-golden-config==1.2.2) (3.5.2)
#12 6.290 Requirement already satisfied: redis>=3.0.0 in /usr/local/lib/python3.7/site-packages (from django-cacheops<6.1,>=6.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.3.4)
#12 6.290 Requirement already satisfied: six>=1.4.0 in /usr/local/lib/python3.7/site-packages (from django-cacheops<6.1,>=6.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.16.0)
#12 6.291 Requirement already satisfied: funcy<2.0,>=1.8 in /usr/local/lib/python3.7/site-packages (from django-cacheops<6.1,>=6.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.17)
#12 6.294 Requirement already satisfied: python-crontab>=2.3.4 in /usr/local/lib/python3.7/site-packages (from django-celery-beat<2.3.0,>=2.2.1->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.6.0)
#12 6.299 Requirement already satisfied: django-picklefield in /usr/local/lib/python3.7/site-packages (from django-constance[database]<2.9.0,>=2.8.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.1)
#12 6.303 Requirement already satisfied: django-appconf in /usr/local/lib/python3.7/site-packages (from django-cryptography<1.1,>=1.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.0.5)
#12 6.303 Requirement already satisfied: cryptography in /usr/local/lib/python3.7/site-packages (from django-cryptography<1.1,>=1.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (37.0.4)
#12 6.316 Requirement already satisfied: django-js-asset in /usr/local/lib/python3.7/site-packages (from django-mptt<0.14.0,>=0.13.4->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.0.0)
#12 6.318 Requirement already satisfied: prometheus-client>=0.7 in /usr/local/lib/python3.7/site-packages (from django-prometheus<2.3.0,>=2.2.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.14.1)
#12 6.325 Requirement already satisfied: rq>=1.2 in /usr/local/lib/python3.7/site-packages (from django-rq<2.6.0,>=2.5.1->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.10.1)
#12 6.351 Requirement already satisfied: inflection>=0.3.1 in /usr/local/lib/python3.7/site-packages (from drf-spectacular[sidecar]<0.23.0,>=0.22.1->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.5.1)
#12 6.351 Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/site-packages (from drf-spectacular[sidecar]<0.23.0,>=0.22.1->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.3.0)
#12 6.352 Requirement already satisfied: uritemplate>=2.0.0 in /usr/local/lib/python3.7/site-packages (from drf-spectacular[sidecar]<0.23.0,>=0.22.1->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.1.1)
#12 6.353 Requirement already satisfied: drf-spectacular-sidecar in /usr/local/lib/python3.7/site-packages (from drf-spectacular[sidecar]<0.23.0,>=0.22.1->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2022.7.1)
#12 6.360 Requirement already satisfied: ruamel.yaml>=0.16.13 in /usr/local/lib/python3.7/site-packages (from drf-yasg[validation]<2.0.0,>=1.20.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.17.21)
#12 6.361 Requirement already satisfied: coreapi>=2.3.3 in /usr/local/lib/python3.7/site-packages (from drf-yasg[validation]<2.0.0,>=1.20.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.3.3)
#12 6.362 Requirement already satisfied: coreschema>=0.0.4 in /usr/local/lib/python3.7/site-packages (from drf-yasg[validation]<2.0.0,>=1.20.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.0.4)
#12 6.363 Requirement already satisfied: swagger-spec-validator>=2.1.0 in /usr/local/lib/python3.7/site-packages (from drf-yasg[validation]<2.0.0,>=1.20.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.7.4)
#12 6.389 Requirement already satisfied: gitdb<5,>=4.0.1 in /usr/local/lib/python3.7/site-packages (from GitPython<3.2.0,>=3.1.27->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.0.9)
#12 6.405 Requirement already satisfied: singledispatch>=3.4.0.3 in /usr/local/lib/python3.7/site-packages (from graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.7.0)
#12 6.406 Requirement already satisfied: text-unidecode in /usr/local/lib/python3.7/site-packages (from graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.3)
#12 6.406 Requirement already satisfied: graphql-core<3,>=2.1.0 in /usr/local/lib/python3.7/site-packages (from graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.3.2)
#12 6.407 Requirement already satisfied: promise>=2.1 in /usr/local/lib/python3.7/site-packages (from graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.3)
#12 6.408 Requirement already satisfied: graphene<3,>=2.1.7 in /usr/local/lib/python3.7/site-packages (from graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.1.9)
#12 6.421 Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/site-packages (from importlib-metadata<5.0,>=4.4->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.8.1)
#12 6.442 Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /usr/local/lib/python3.7/site-packages (from jsonschema<4.8.0,>=4.7.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.18.1)
#12 6.443 Requirement already satisfied: importlib-resources>=1.4.0 in /usr/local/lib/python3.7/site-packages (from jsonschema<4.8.0,>=4.7.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.9.0)
#12 6.443 Requirement already satisfied: attrs>=17.4.0 in /usr/local/lib/python3.7/site-packages (from jsonschema<4.8.0,>=4.7.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (22.1.0)
#12 6.491 Collecting nornir-utils<1,>=0
#12 6.506   Downloading nornir_utils-0.2.0-py3-none-any.whl (15 kB)
#12 6.533 Collecting pynautobot<2.0.0,>=1.0.1
#12 6.551   Downloading pynautobot-1.2.2-py3-none-any.whl (31 kB)
#12 6.573 Collecting nornir-netmiko<1,>=0
#12 6.593   Downloading nornir_netmiko-0.2.0-py3-none-any.whl (11 kB)
#12 6.597 Requirement already satisfied: requests<3.0.0,>=2.25.1 in /usr/local/lib/python3.7/site-packages (from nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (2.27.1)
#12 6.627 Collecting nornir-jinja2<1,>=0
#12 6.642   Downloading nornir_jinja2-0.2.0-py3-none-any.whl (7.2 kB)
#12 6.666 Collecting nornir<4.0.0,>=3.0.0
#12 6.679   Downloading nornir-3.3.0-py3-none-any.whl (30 kB)
#12 6.702 Collecting nornir-napalm<1,>=0
#12 6.718   Downloading nornir_napalm-0.3.0-py3-none-any.whl (10 kB)
#12 6.766 Requirement already satisfied: social-auth-core>=4.1.0 in /usr/local/lib/python3.7/site-packages (from social-auth-app-django<5.1.0,>=5.0.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.3.0)
#12 6.811 Requirement already satisfied: prompt-toolkit in /usr/local/lib/python3.7/site-packages (from click-repl>=0.2.0->celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.0.30)
#12 6.815 Requirement already satisfied: itypes in /usr/local/lib/python3.7/site-packages (from coreapi>=2.3.3->drf-yasg[validation]<2.0.0,>=1.20.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.2.0)
#12 6.825 Requirement already satisfied: smmap<6,>=3.0.1 in /usr/local/lib/python3.7/site-packages (from gitdb<5,>=4.0.1->GitPython<3.2.0,>=3.1.27->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.0.0)
#12 6.837 Requirement already satisfied: aniso8601<=7,>=3 in /usr/local/lib/python3.7/site-packages (from graphene<3,>=2.1.7->graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (7.0.0)
#12 6.838 Requirement already satisfied: graphql-relay<3,>=2 in /usr/local/lib/python3.7/site-packages (from graphene<3,>=2.1.7->graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.0.1)
#12 6.848 Requirement already satisfied: rx<2,>=1.6 in /usr/local/lib/python3.7/site-packages (from graphql-core<3,>=2.1.0->graphene-django<2.16.0,>=2.15.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.6.1)
#12 6.878 Requirement already satisfied: cached-property in /usr/local/lib/python3.7/site-packages (from kombu<6.0,>=5.2.3->celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.5.2)
#12 6.878 Requirement already satisfied: amqp<6.0.0,>=5.0.9 in /usr/local/lib/python3.7/site-packages (from kombu<6.0,>=5.2.3->celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (5.1.1)
#12 6.888 Requirement already satisfied: mypy_extensions<0.5.0,>=0.4.1 in /usr/local/lib/python3.7/site-packages (from nornir<4.0.0,>=3.0.0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (0.4.3)
#12 6.933 Collecting napalm<5,>=4
#12 6.945   Downloading napalm-4.0.0-py2.py3-none-any.whl (266 kB)
#12 6.959      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 266.2/266.2 kB 22.2 MB/s eta 0:00:00
#12 6.993 Collecting netmiko<5.0.0,>=4.0.0
#12 7.010   Downloading netmiko-4.1.2-py3-none-any.whl (196 kB)
#12 7.020      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 196.8/196.8 kB 24.2 MB/s eta 0:00:00
#12 7.045 Collecting textfsm==1.1.2
#12 7.063   Downloading textfsm-1.1.2-py2.py3-none-any.whl (44 kB)
#12 7.067      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.7/44.7 kB 11.8 MB/s eta 0:00:00
#12 7.073 Requirement already satisfied: future in /usr/local/lib/python3.7/site-packages (from textfsm==1.1.2->nornir-netmiko<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (0.18.2)
#12 7.109 Collecting colorama<0.5.0,>=0.4.3
#12 7.130   Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
#12 7.175 Requirement already satisfied: deprecated>=1.2.3 in /usr/local/lib/python3.7/site-packages (from redis>=3.0.0->django-cacheops<6.1,>=6.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.2.13)
#12 7.175 Requirement already satisfied: async-timeout>=4.0.2 in /usr/local/lib/python3.7/site-packages (from redis>=3.0.0->django-cacheops<6.1,>=6.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (4.0.2)
#12 7.187 Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/site-packages (from requests<3.0.0,>=2.25.1->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (2022.6.15)
#12 7.187 Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.7/site-packages (from requests<3.0.0,>=2.25.1->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (2.0.12)
#12 7.188 Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.7/site-packages (from requests<3.0.0,>=2.25.1->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (1.26.11)
#12 7.189 Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.7/site-packages (from requests<3.0.0,>=2.25.1->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (3.3)
#12 7.202 Requirement already satisfied: ruamel.yaml.clib>=0.2.6 in /usr/local/lib/python3.7/site-packages (from ruamel.yaml>=0.16.13->drf-yasg[validation]<2.0.0,>=1.20.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.2.6)
#12 7.230 Requirement already satisfied: oauthlib>=1.0.3 in /usr/local/lib/python3.7/site-packages (from social-auth-core>=4.1.0->social-auth-app-django<5.1.0,>=5.0.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.2.0)
#12 7.231 Requirement already satisfied: defusedxml>=0.5.0rc1 in /usr/local/lib/python3.7/site-packages (from social-auth-core>=4.1.0->social-auth-app-django<5.1.0,>=5.0.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.7.1)
#12 7.231 Requirement already satisfied: python3-openid>=3.0.10 in /usr/local/lib/python3.7/site-packages (from social-auth-core>=4.1.0->social-auth-app-django<5.1.0,>=5.0.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (3.2.0)
#12 7.232 Requirement already satisfied: requests-oauthlib>=0.6.1 in /usr/local/lib/python3.7/site-packages (from social-auth-core>=4.1.0->social-auth-app-django<5.1.0,>=5.0.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.3.1)
#12 7.232 Requirement already satisfied: PyJWT>=2.0.0 in /usr/local/lib/python3.7/site-packages (from social-auth-core>=4.1.0->social-auth-app-django<5.1.0,>=5.0.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.4.0)
#12 7.261 Requirement already satisfied: cffi>=1.12 in /usr/local/lib/python3.7/site-packages (from cryptography->django-cryptography<1.1,>=1.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.15.1)
#12 7.365 Requirement already satisfied: pycparser in /usr/local/lib/python3.7/site-packages (from cffi>=1.12->cryptography->django-cryptography<1.1,>=1.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (2.21)
#12 7.399 Requirement already satisfied: wrapt<2,>=1.10 in /usr/local/lib/python3.7/site-packages (from deprecated>=1.2.3->redis>=3.0.0->django-cacheops<6.1,>=6.0->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (1.14.1)
#12 7.428 Requirement already satisfied: scp in /usr/local/lib/python3.7/site-packages (from napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (0.14.4)
#12 7.429 Requirement already satisfied: junos-eznc>=2.6.3 in /usr/local/lib/python3.7/site-packages (from napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (2.6.5)
#12 7.430 Requirement already satisfied: paramiko>=2.6.0 in /usr/local/lib/python3.7/site-packages (from napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (2.11.0)
#12 7.431 Requirement already satisfied: lxml>=4.3.0 in /usr/local/lib/python3.7/site-packages (from napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (4.6.5)
#12 7.431 Requirement already satisfied: pyeapi>=0.8.2 in /usr/local/lib/python3.7/site-packages (from napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (0.8.4)
#12 7.432 Requirement already satisfied: ncclient in /usr/local/lib/python3.7/site-packages (from napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (0.6.13)
#12 7.433 Requirement already satisfied: setuptools>=38.4.0 in /usr/local/lib/python3.7/site-packages (from napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (57.5.0)
#12 7.464 Collecting ttp-templates
#12 7.478   Downloading ttp_templates-0.3.1-py3-none-any.whl (61 kB)
#12 7.483      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 13.9 MB/s eta 0:00:00
#12 7.507 Collecting ttp
#12 7.522   Downloading ttp-0.9.1-py2.py3-none-any.whl (84 kB)
#12 7.528      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.7/84.7 kB 16.0 MB/s eta 0:00:00
#12 7.537 Requirement already satisfied: tenacity in /usr/local/lib/python3.7/site-packages (from netmiko<5.0.0,>=4.0.0->nornir-netmiko<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (8.0.1)
#12 7.537 Requirement already satisfied: ntc-templates>=2.0.0 in /usr/local/lib/python3.7/site-packages (from netmiko<5.0.0,>=4.0.0->nornir-netmiko<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (3.0.0)
#12 7.538 Requirement already satisfied: pyserial in /usr/local/lib/python3.7/site-packages (from netmiko<5.0.0,>=4.0.0->nornir-netmiko<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (3.5)
#12 7.624 Requirement already satisfied: wcwidth in /usr/local/lib/python3.7/site-packages (from prompt-toolkit->click-repl>=0.2.0->celery<5.3.0,>=5.2.7->nautobot>=1.4.0->nautobot-golden-config==1.2.2) (0.2.5)
#12 7.634 Requirement already satisfied: transitions in /usr/local/lib/python3.7/site-packages (from junos-eznc>=2.6.3->napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (0.8.11)
#12 7.634 Requirement already satisfied: yamlordereddictloader in /usr/local/lib/python3.7/site-packages (from junos-eznc>=2.6.3->napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (0.4.0)
#12 7.672 Requirement already satisfied: bcrypt>=3.1.3 in /usr/local/lib/python3.7/site-packages (from paramiko>=2.6.0->napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (3.2.2)
#12 7.673 Requirement already satisfied: pynacl>=1.0.1 in /usr/local/lib/python3.7/site-packages (from paramiko>=2.6.0->napalm<5,>=4->nornir-napalm<1,>=0->nornir-nautobot<3.0.0,>=2.2.0->nautobot-plugin-nornir>=1.0.0->nautobot-golden-config==1.2.2) (1.5.0)
#12 7.830 Building wheels for collected packages: nautobot-golden-config
#12 7.832   Building editable for nautobot-golden-config (pyproject.toml): started
#12 7.992   Building editable for nautobot-golden-config (pyproject.toml): finished with status 'done'
#12 7.993   Created wheel for nautobot-golden-config: filename=nautobot_golden_config-1.2.2-py3-none-any.whl size=4238 sha256=4a2808d0542328a0d2fe1e34844cc6181bf881fbab7ef5e61e1d3bab83c77599
#12 7.993   Stored in directory: /tmp/pip-ephem-wheel-cache-78lbhaqb/wheels/42/f2/10/c79060ae8713a08dd7c566c1e35c7b11ad3fdbfee08948f096
#12 7.995 Successfully built nautobot-golden-config
#12 8.208 Installing collected packages: ttp, textfsm, ordered-set, numpy, kiwisolver, fonttools, cycler, colorama, ttp-templates, pynautobot, nornir, matplotlib, deepdiff, nornir-utils, nornir-jinja2, django-pivot, netmiko, nornir-netmiko, napalm, nornir-napalm, nornir-nautobot, nautobot-plugin-nornir, nautobot-golden-config
#12 8.259   Attempting uninstall: textfsm
#12 8.260     Found existing installation: textfsm 1.1.3
#12 8.263     Uninstalling textfsm-1.1.3:
#12 8.271       Successfully uninstalled textfsm-1.1.3
#12 10.71   Attempting uninstall: netmiko
#12 10.71     Found existing installation: netmiko 3.4.0
#12 10.75     Uninstalling netmiko-3.4.0:
#12 10.82       Successfully uninstalled netmiko-3.4.0
#12 10.94   Attempting uninstall: napalm
#12 10.94     Found existing installation: napalm 3.4.1
#12 10.96     Uninstalling napalm-3.4.1:
#12 10.99       Successfully uninstalled napalm-3.4.1
#12 11.15 Successfully installed colorama-0.4.6 cycler-0.11.0 deepdiff-5.8.1 django-pivot-1.9.0 fonttools-4.38.0 kiwisolver-1.4.4 matplotlib-3.5.3 napalm-4.0.0 nautobot-golden-config-1.2.2 nautobot-plugin-nornir-1.0.0 netmiko-4.1.2 nornir-3.3.0 nornir-jinja2-0.2.0 nornir-napalm-0.3.0 nornir-nautobot-2.3.0 nornir-netmiko-0.2.0 nornir-utils-0.2.0 numpy-1.21.6 ordered-set-4.1.0 pynautobot-1.2.2 textfsm-1.1.2 ttp-0.9.1 ttp-templates-0.3.1
#12 11.15 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 11.15
#12 11.15 [notice] A new release of pip available: 22.2.2 -> 22.3.1
#12 11.15 [notice] To update, run: pip install --upgrade pip
#12 DONE 11.4s

#13 [ 9/11] RUN poetry install --only dev
#13 0.505 Skipping virtualenv creation, as specified in config file.
#13 2.024 Installing dependencies from lock file
#13 3.736
#13 3.736 Package operations: 6 installs, 7 updates, 0 removals
#13 3.736
#13 3.737   • Updating jinja2 (3.0.3 -> 3.1.2)
#13 4.991   • Updating setuptools (57.5.0 -> 65.5.1)
#13 6.758   • Installing pbr (5.10.0)
#13 7.756   • Updating pylint (2.14.5 -> 2.13.9)
#13 9.031   • Updating pytz (2022.1 -> 2022.2.1)
#13 10.17   • Updating django (3.2.15 -> 3.2.16)
#13 14.36   • Updating gitpython (3.1.27 -> 3.1.29)
#13 15.40   • Installing snowballstemmer (2.2.0)
#13 16.44   • Installing stevedore (3.5.0)
#13 17.43   • Installing bandit (1.7.4)
#13 18.53   • Updating mkdocs-material (8.4.2 -> 8.3.9)
#13 22.69   • Installing pydocstyle (6.1.1)
#13 23.64   • Installing yamllint (1.27.1)
#13 27.40
#13 27.40 Installing the current project: nautobot-golden-config (1.2.2)
#13 DONE 27.7s

#14 [10/11] RUN poetry run python -c "import nautobot; print(nautobot.__version__);"
#14 0.583 Skipping virtualenv creation, as specified in config file.
#14 0.644 1.4.2 # <======= Nautobot version from inside Python ==========================================================
#14 DONE 0.7s

#15 [11/11] COPY development/nautobot_config.py /opt/nautobot/nautobot_config.py
#15 DONE 0.1s

#16 exporting to image
#16 exporting layers
#16 exporting layers 4.0s done
#16 exporting manifest sha256:910fa2a2e3689cf30d2ccd42c2c6bf4d2cb1219a78618fd5abd554a25378c625 done
#16 exporting config sha256:29e87fc6d9b63cf5f2c9efc2df8d341df2c9a52f4e1c34563156e6dc16edb90d done
#16 naming to docker.io/nautobot-golden-config/nautobot:1.4.2-py3.7 done
#16 unpacking to docker.io/nautobot-golden-config/nautobot:1.4.2-py3.7
#16 unpacking to docker.io/nautobot-golden-config/nautobot:1.4.2-py3.7 3.4s done
#16 DONE 7.5s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
```